src/test/java/com/google/sps/UserTripServletTest.java

```diff
@@ -74,94 +89,167 @@ public void tearDown() {
+    tripEntity.setProperty(Trip.TRIP_NAME, TRIP_NAME);
+    tripEntity.setProperty(Trip.DESTINATION_NAME, DESTINATION_NAME);
+    tripEntity.setProperty(Trip.IMAGE_SRC, IMAGE_SRC);
+    tripEntity.setProperty(Trip.START_DATE, TRIP_DAY_OF_TRAVEL);
+    tripEntity.setProperty(Trip.END_DATE, TRIP_DAY_OF_TRAVEL);
+    datastore.put(tripEntity);
+
+    // Add a second Trip to Datastore with the User Entity Key.
+    Entity tripEntity2 = new Entity(Trip.TRIP, userEntityKey);
+    tripEntity2.setProperty(Trip.TRIP_NAME, TRIP_NAME_2);
+    tripEntity2.setProperty(Trip.DESTINATION_NAME, DESTINATION_NAME_2);
+    tripEntity2.setProperty(Trip.IMAGE_SRC, IMAGE_SRC_2);
+    tripEntity2.setProperty(Trip.START_DATE, TRIP_DAY_OF_TRAVEL_2);
+    tripEntity2.setProperty(Trip.END_DATE, TRIP_DAY_OF_TRAVEL_2);
+    datastore.put(tripEntity2);
+
+    // Return the User Entity Key, used to Query for the Trip Entity.
+    return userEntityKey;
+  }
+
   @Test
-  public void testDoGetResponseLoggedIn() throws Exception {
+  public void testWriteTripsToFileResponseNoTrip() throws Exception {
     // Mock request and response.  
     HttpServletRequest requestMock = mock(HttpServletRequest.class);    
     HttpServletResponse responseMock = mock(HttpServletResponse.class);
-    
-    // Create Trip Entity properties.
-    final String tripName = "Family Vacation";
-    final String destinationName = "Island of Hawai'i";
-    final String tripDayOfTravel = "2020-07-17";
-    final String photoSrc = "../images/placeholder_image.png";
-
-    // Mock UserService methods as logged-in user.
-    UserService userServiceMock = mock(UserService.class);
-    when(userServiceMock.isUserLoggedIn()).thenReturn(true);
-
-    // This is the User object from Google Appengine (full path given to avoid
-    // confusion with local User.java file).
-    when(userServiceMock.getCurrentUser()).thenReturn(
-        new com.google.appengine.api.users.User(EMAIL, AUTH_DOMAIN));
-    when(userServiceMock.createLogoutURL(AuthServlet.redirectUrl)).thenReturn(LOGOUT_URL);
-
-    // PowerMock static getUserService() method, which is used to get the user.
-    PowerMockito.mockStatic(UserServiceFactory.class);
-    when(UserServiceFactory.getUserService()).thenReturn(userServiceMock);
-
-    // Run storeTripEntity(...), with the User logged in (so trip is stored).
-    tripServlet.storeTripEntity(responseMock,
-      tripName, destinationName, tripDayOfTravel, photoSrc);
 
     // Create writers to pass into the mock response object.
     StringWriter stringWriter = new StringWriter();
     PrintWriter writer = new PrintWriter(stringWriter);
     when(responseMock.getWriter()).thenReturn(writer);
 
-    // Run the UserServlet doGet(...) method with request and response mocks. 
-    userTripServlet.doGet(requestMock, responseMock); 
+    // Get the User Entity Key, and set up the database using the helper method.
+    Key userEntityKey = addCurrentUserAndNoTripDatastore();
```

Comment:
Though it may seem counter-intuitive, in tests it's generally discouraged to use helper methods unless something is being done a bunch of times (and in some languages at Google, such as c++, even that is discouraged). The reasoning is that with tests it's important that you understand exactly what's happening in the test, to ensure that nothing is being obfuscated. When people read helper methods in tests, it's easy to just glide your eyes over them instead of hunting down what's happening in that method.

So though it makes the test itself longer, you probably should stick these helper methods back into the tests they come from.

I know it goes against what we do in regular code, so let me know if you're still confused by this

Reply:
I think I would be cool with you making a populateTripEntity helper method, since that is pretty simple setting code, and will be re-used multiple places, and is easy to digest

Reply:
The `populateTripEntity` helper method would be the same `buildEntity` method in the `Trip.java` class. Can I just use that @bheff-work?

Reply:
Actually, I didn't want to over-complicate things, so I just included all of this code in each test. There are only three tests here, so it's not repeated too many times.